### PR TITLE
RavenDB-17732 move native DLLs to runtime folder inside NuGet package

### PR DESCRIPTION
**Pull request by ppekrol**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17732

### Additional description

This moves the libzstd libraries to runtimes folder according to https://docs.microsoft.com/en-us/nuget/create-packages/supporting-multiple-target-frameworks#architecture-specific-folders and https://docs.microsoft.com/en-us/dotnet/core/rid-catalog

The change was tested using .NET 6.0 SDK and following projects:
- netstandard2.0 library
- net6.0 app
- net471 app

### Type of change

- Optimization

### How risky is the change?

- Moderate 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: scripts/nuget.ps1 [hunk 1]
```diff
@@ -38,9 +38,7 @@ function BuildEmbeddedNuget ($projectDir, $outDir, $serverSrcDir, $studioZipPath
     $EMBEDDED_OUT_DIR = [io.path]::combine($outDir, "RavenDB.Embedded")
     $EMBEDDED_BUILD_OUT_DIR = [io.path]::combine($EMBEDDED_OUT_DIR, "build")
     $EMBEDDED_SERVER_OUT_DIR = [io.path]::combine($EMBEDDED_OUT_DIR, "contentFiles", "any", "any")
-    
-    $LIB_ZSTD_DIR = [io.path]::combine($projectDir, "libs", "libzstd", "*")
```
Comment (ayende) [inline scripts/nuget.ps1:42]: Don't we have to do the same for Libsodium?

Comment (ppekrol) [reply scripts/nuget.ps1:42]: No, libsodium is not used by Client

**Comment by ryanbuening** [comment]
We are still seeing this error (on 5.3.2) when running `dotnet publish` locally and in our Azure DevOps pipeline. Is this fix in 5.3.2?

```
PS C:\dev\Claims\src\Claims.Web> dotnet publish
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored C:\dev\Claims\src\DispatchR\DispatchR.csproj (in 279 ms).
  Restored C:\dev\Claims\src\Claims.Core\Claims.Core.csproj (in 360 ms).
  Restored C:\dev\Claims\src\Claims.Web\Claims.Web.csproj (in 519 ms).
  DispatchR -> C:\dev\Claims\src\DispatchR\bin\Debug\net6.0\DispatchR.dll
  Claims.Core -> C:\dev\Claims\src\Claims.Core\bin\Debug\net6.0\Claims.Core.dll
  Claims.Web -> C:\dev\Claims\src\Claims.Web\bin\Debug\net6.0\Claims.Web.dll
C:\Program Files\dotnet\sdk\6.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ConflictResolution.targets(112,5): error NETSDK1152: Found multiple publish output files with the same relative path: C:\Users\user\.nuget\packages\ravendb.client\5.3.2\contentFiles\any\any\libzstd.arm.32.so, C:\Users\user\.nuget\packages\ravendb.embedded\5.3.2\contentFiles\any\any\libzstd.arm.32.so, C:\Users\user\.nuget\packages\ravendb.client\5.3.2\contentFiles\any\any\libzstd.arm.64.so, C:\Users\user\.nuget\packages\ravendb.embedded\5.3.2\contentFiles\any\any\libzstd.arm.64.so, C:\Users\user\.nuget\packages\ravendb.client\5.3.2\contentFiles\any\any\libzstd.linux.x64.so, C:\Users\user\.nuget\packages\ravendb.embedded\5.3.2\contentFiles\any\any\libzstd.linux.x64.so, C:\Users\user\.nuget\packages\ravendb.client\5.3.2\contentFiles\any\any\libzstd.mac.x64.dylib, C:\Users\user\.nuget\packages\ravendb.embedded\5.3.2\contentFiles\any\any\libzstd.mac.x64.dylib, C:\Users\user\.nuget\packages\ravendb.client\5.3.2\contentFiles\any\any\libzstd.win.x64.dll, C:\Users\user\.nuget\packages\ravendb.embedded\5.3.2\contentFiles\any\any\libzstd.win.x64.dll, C:\Users\user\.nuget\packages\ravendb.client\5.3.2\contentFiles\any\any\libzstd.win.x86.dll, C:\Users\user\.nuget\packages\ravendb.embedded\5.3.2\contentFiles\any\any\libzstd.win.x86.dll. [C:\dev\Claims\src\Claims.Web\Claims.Web.csproj]
```

**Comment by arekpalinski** [comment]
No, this fix will be included in the next release (beginning of Feb).

**Comment by arekpalinski** [comment]
Although it's already included in the nightly version: https://myget.org/gallery/ravendb

**Pull request merged**